lib/zclient.c

```diff
@@ -896,6 +956,14 @@ static int zapi_nexthop_cmp(const void *item1, const void *item2)
 		return ret;
 
 	ret = zapi_nexthop_labels_cmp(next1, next2);
+	if (ret != 0)
+		return ret;
+
+	ret = zapi_nexthop_seg6local_cmp(next1, next2);
+	if (ret != 0)
+		return ret;
+
+	ret = zapi_nexthop_seg6_cmp(next1, next2);
```

Comment:
can't we just have a single `seg6_cmp` function - do we need two?

Reply:
I integrated zapi_nexthop_seg6local_cmp function and zapi_nexthop_seg6_cmp function into one function as your comment.

And I changed the name of the function to zapi_nexthop_srv6_cmp because I felt that the name seg6_cmp would cause confusion.

This is because seg6 (i.e. SRv6 Transit Behavior) and seg6local (SRv6 Local Function Behavior) are different network programming primitives.

If both of them have more complex extensions in the future, I think it will be less confusing to remove the integration of these functions and make them separate functions as before. And I agree with your point since it is simple at the moment.